src/github.com/matrix-org/dendrite/clientapi/writers/createroom.go

```diff
@@ -34,9 +75,47 @@ func CreateRoom(req *http.Request) util.JSONResponse {
 	if resErr != nil {
 		return *resErr
 	}
+	// TODO: apply rate-limit
+
+	if resErr = r.Validate(); resErr != nil {
+		return *resErr
+	}
+
+	// TODO: visibility/presets/raw initial state/creation content
+
+	hostname := "localhost"
+	roomID := fmt.Sprintf("!%s:%s", util.RandomString(16), hostname)
```

Comment:
Is util.RandomString a good choice for generating globally unique IDs?
Is there a way to control the output of util.RandomString for unit testing?